common/src/main/java/tachyon/dag/DAG.java

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
I would avoid using abbreviations for class name, so I would call this class DirectedAcyclicGraph instead of DAG. This is similar to how we use FileSystem instead of FS in our class names.